control-plane-operator/controllers/hostedcontrolplane/v2/assets/dns-operator/deployment.yaml

```
@@ -0,0 +1,52 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dns-operator
+  namespace: HCP_NAMESPACE
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: dns-operator
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
```

missing annotation 
```suggestion
    metadata:
      annotations:
        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
```

from v1 code https://github.com/openshift/hypershift/blob/2753dcc8cfb7591a61d64d26766526f844cebffa/control-plane-operator/controllers/hostedcontrolplane/dnsoperator/dnsoperator.go#L63